### PR TITLE
feat: added a login page and authentication using passportjs

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
feat: add support for gm4c rbac authentication
feat: added endpoint to get the login options. added gm4c strategy
chore: upgrade to nodejs 14 for certain packages to work properly

Signed-off-by: Harikrishnan Balagopal <harikrishmenon@gmail.com>

### CODE REVIEW

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Looks good. Thanks for the oauth support. Looking forward to getting it integrated.

File: .dockerignore [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #   Copyright IBM Corporation 2020
 #
-#   Licensed under the Apache License, Version 2.0 (the "License");
 #   you may not use this file except in compliance with the License.
+#   Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (ashokponkumar) [inline .dockerignore:4]: This seems to have gotten moved by mistake.

File: .gitignore [hunk 2]
```diff
@@ -23,3 +23,18 @@ storybook-static
 .env
 wksps/
```
Comment (ashokponkumar) [inline .gitignore:24]: Do we need to check-in the .yarn directory?

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
This looks good. Few minor questions. One we are done with it, we can merge.

File: .gitignore (hunk 2 above)
Comment (HarikrishnanBalagopal) [reply .gitignore:24]: yeah parts of it need to be checked in https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored

File: .dockerignore (hunk 1 above)
Comment (HarikrishnanBalagopal) [reply .dockerignore]: oops. fixed it.

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**